src/clj_kondo/impl/linters/keys.clj

```diff
@@ -5,21 +5,15 @@
    [clj-kondo.impl.utils :refer [node->line tag]]
    [clojure.string :as string]))
 
-(defn- ensure-no-nils
-  "Returns nil if the collection contains a nil, else returns coll.
-  Doesn't search for nested nils."
-  [coll]
-  (cond
-    (set? coll) (if (contains? coll nil)
-                  nil
-                  coll)
-    (sequential? coll) (if (contains? (set coll) nil)
-                         nil
-                         coll)
-    (map? coll) (if (contains? (set (flatten (seq coll))) nil)
-                  nil
-                  coll)
-    :else coll))
+(defn- map-without-nils
+  "Like map, but returns nil if any mapped value is nil."
+  [f coll]
+  (reduce (fn [acc v]
```

Comment:
Might want to use a transient here, like in `mapv`:

```
([f coll]
     (-> (reduce (fn [v o] (conj! v (f o))) (transient []) coll)
         persistent!))
```

Reply:
Addressed this, thanks for the tip.